server/ServerApp.cpp

```diff
@@ -3333,12 +3333,13 @@ void ServerApp::PostCombatProcessTurns() {
     // Consume distributed resources to planets and on queues, create new
     // objects for completed production and give techs to empires that have
     // researched them
+    std::vector<std::pair<Empire *,std::vector<std::string>>> empires_innovations;
```

Comment:
better to store `int` empire ID than the pointer, but if you store a pointer, write it `Empire*, std::vector<std::string>` with no space before `*` and a space after `,`

Reply:
Trying to understand your reasoning:
This is local vector and I need a reference/pointer later in the same function.
Doing a lookup using the empire ID would be better because the empire could be deleted in between(?)

Reply:
It's probably fine here, but in general I prefer to keep track of IDs rather than raw pointers to stuff. Perhaps someone could be inspired by this bit of code to do something more problematic elsewhere.